signer/src/api/info.rs

```diff
@@ -0,0 +1,505 @@
+                            block_height: local_stacks_chain_tip.block_height,
+                        });
+                    }
+                    Ok(None) => {
+                        tracing::debug!("no local stacks tip found in the database.");
+                    }
+                    Err(e) => {
+                        tracing::error!("error reading local Stacks tip from the database: {}", e);
+                    }
+                }
+            }
+            Ok(None) => {
+                tracing::debug!("no local bitcoin tip found in the database.");
+            }
+            Err(e) => {
+                tracing::error!("error reading bitcoin tip from the database: {}", e);
+            }
+        }
+    }
+
+    /// Populates the Bitcoin node tip information from the provided Bitcoin
+    /// client. This uses a combination of `getblockchaininfo` and
+    /// `getnetworkinfo` RPC calls to populate the information.
+    async fn populate_bitcoin_node_info(&mut self, bitcoin_client: &impl BitcoinInteract) {
+        match bitcoin_client.get_blockchain_info().await {
+            Ok(info) => {
+                self.bitcoin.node_chain = Some(info.chain.to_string());
+                self.bitcoin.node_tip = Some(ChainTipInfo {
+                    block_hash: info.best_block_hash.into(),
+                    block_height: info.blocks,
+                });
+            }
+            Err(e) => {
+                tracing::error!("error getting bitcoin node blockchain info: {}", e);
+            }
+        }
+
+        match bitcoin_client.get_network_info().await {
+            Ok(info) => {
+                self.bitcoin.node_version = Some(info.version);
+                self.bitcoin.node_subversion = Some(info.subversion);
+            }
+            Err(e) => {
+                tracing::error!("error getting bitcoin node network info: {}", e);
+            }
+        }
+    }
+
+    /// Populates the Stacks node tip information from the provided Stacks client.
+    /// This uses the `/v2/info` RPC endpoint to populate the information.
+    async fn populate_stacks_node_info(&mut self, stacks_client: &impl StacksInteract) {
+        match stacks_client.get_node_info().await {
+            Ok(node_info) => {
+                self.stacks.node_tip = Some(ChainTipInfo {
+                    block_hash: node_info.stacks_tip.0.into(),
+                    block_height: node_info.stacks_tip_height,
+                });
+                self.stacks.node_bitcoin_block_height = Some(node_info.burn_block_height);
+                self.stacks.node_version = Some(node_info.server_version);
+            }
+            Err(e) => {
+                tracing::error!("error getting stacks node tip: {}", e);
+            }
+        }
+    }
+
+    /// Populates the DKG information from the provided storage.
+    async fn populate_dkg_info(&mut self, storage: &impl DbRead, config: &Settings) {
```

Comment:
I was going to say that we should fetch the aggregate key from the stacks node, but maybe that isn't necessary. I don't have a strong feeling about it so I'm not suggesting one or the other.

Reply:
Well, why not both? 😄 That let's us see if there's a diff between the contract and the local node, i.e. maybe rotate keys is failing.

[a8582d9](https://github.com/stacks-network/sbtc/pull/1251/commits/a8582d963a087e7a1aa0fc58299183e077e4b195)